modules/token/keeper/msg_server.go

```diff
@@ -325,6 +323,21 @@ func (m msgServer) SwapFromERC20(goCtx context.Context, msg *v1.MsgSwapFromERC20
 }
 
 // SwapToERC20 implements v1.MsgServer.
-func (m msgServer) SwapToERC20(context.Context, *v1.MsgSwapToERC20) (*v1.MsgSwapToERC20Response, error) {
-	panic("unimplemented")
+func (m msgServer) SwapToERC20(goCtx context.Context, msg *v1.MsgSwapToERC20) (*v1.MsgSwapToERC20Response, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return nil, err
+	}
+
+	bz, err := hex.DecodeString(msg.Receiver)
+	if err != nil {
```

Comment:
receiver address already validated in `ValidateBasic`, it's unnecessary to recheck; however, it doesn hurt to do it again here, your call.